src/main/scala/com/ubirch/messagesigner/MessageSignerMicroservice.scala

```diff
@@ -37,8 +42,12 @@ class MessageSignerMicroservice(
       signers(curve)
     }
 
-    signer.sign(record).toProducerRecord(onlyOutputTopic)
+    signer.sign(record).toProducerRecord(getOutputTopic(record))
+  }
 
+  private def getOutputTopic(r: ConsumerRecord[String, MessageEnvelope]): String = {
+    if (r.existsHeader(GATEWAY_TYPE_HEADER) && r.findHeader(GATEWAY_TYPE_HEADER).get == MQTT_KEY) mqttTopic
```

Comment:
I think we can have a more compacted and faster predicate if we do
`if (r.findHeader(GATEWAY_TYPE_HEADER).contains(MQTT_KEY)) mqttTopic`

Reply:
if findFinder returns null, this won't produce any problems?
I will test.

Reply:
fixed it.